simplq/src/components/pages/Admin/index.jsx

```diff
@@ -15,52 +14,56 @@ import AddMember from './AddMember';
 const TIMEOUT = 10000;
 let timeoutId;
 
-export default () => {
+export default (props) => {
   const dispatch = useDispatch();
-  const queueId = useSelector((state) => state.appReducer.queueId);
-  const queueName = useSelector((state) => state.appReducer.queueName);
-
-  if (!queueId) {
-    // If queue id is not here, most probably his session storage got cleared. This can be solved only with proper auth.
-    return <PageNotFound />;
-  }
+  const queueId = props.match.params.queueId;
 
   dispatch(progressCreationStep(1));
 
   const [items, setItems] = useState();
-
+  const [queueName, setQueueName] = useState();
   const update = useCallback(() => {
     clearTimeout(timeoutId);
-    if (queueId) {
-      QueueService.get(queueId)
-        .then((data) => {
-          setItems(data.tokens);
-          timeoutId = setTimeout(update, TIMEOUT);
-        })
-        .catch((err) => {
-          handleApiErrors(err);
-          timeoutId = setTimeout(update, TIMEOUT);
-        });
-    }
+    QueueService.get(queueId)
+      .then((data) => {
+        setItems(data.tokens);
+        setQueueName(data.queueName);
+        timeoutId = setTimeout(update, TIMEOUT);
+      })
+      .catch((err) => {
+        handleApiErrors(err);
+        timeoutId = setTimeout(update, TIMEOUT);
+      });
   }, [queueId]);
```

Comment:
Did you remove the queueId condition because this only fires if queueId gets a valid value?

Reply:
It was some hack I did because queueId was coming from localstorage, and it was coming as null. Now it comes from url. So I can expect it to be present.